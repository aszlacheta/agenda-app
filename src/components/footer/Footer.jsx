import React from 'react';
import { useTranslation } from 'react-i18next';
import './Footer.scss';

export default function Footer() {
    const year = new Date().getFullYear();
    const { t } = useTranslation();


    return <div className="footer" data-cy="footer"><span>{t('footer.label', {year})}</span></div>
}