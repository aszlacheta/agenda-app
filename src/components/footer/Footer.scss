@use '../variables.scss';

.footer {
    display: flex;
    flex-direction: column;
    justify-content: center;
    height: variables.$footerHeight;
    max-height: variables.$footerMaxHeight;
    padding: variables.$footerPadding;

    background: variables.$footerBackground;

    text-align: right;
    text-transform: uppercase;
    font-size: variables.$footerTextFontSize;
    letter-spacing: variables.$footerTextLetterSpacing;
}