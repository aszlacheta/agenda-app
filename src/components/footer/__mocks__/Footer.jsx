import React from 'react';

const Footer = () => <div className="mocked-footer" />

module.exports = Footer;