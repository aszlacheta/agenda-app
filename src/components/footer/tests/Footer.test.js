import React from 'react';
import renderer from 'react-test-renderer';
import Footer from '../Footer'; 
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() })

describe('Footer', () => {
    it('check if renders properly', () => {
        const tree = renderer
            .create(<Footer />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('check if renders current year', () => {
        const footer = shallow(<Footer />);
        const currentYear = new Date().getFullYear();

        expect(footer.text()).toEqual(`All rights reserved @ ${currentYear}`);
    })
});