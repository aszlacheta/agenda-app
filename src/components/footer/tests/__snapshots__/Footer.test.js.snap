// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer check if renders properly 1`] = `
<div
  className="footer"
  data-cy="footer"
>
  <span>
    footer.label, 2022
  </span>
</div>
`;
